File: app/src/main/java/com/nakano/stampcardmvvm/viewModel/AuthViewModel.java
package com.nakano.stampcardmvvm.viewModel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.google.firebase.auth.AuthCredential;
import com.nakano.stampcardmvvm.model.model.UserFirebase;
import com.nakano.stampcardmvvm.model.repository.AuthRepository;

public class AuthViewModel extends AndroidViewModel {
    private AuthRepository authRepository;
    public LiveData<UserFirebase> authenticatedUserLiveData;
    public LiveData<UserFirebase> createdUserLiveData;

    public AuthViewModel(Application application) {
        super(application);
        authRepository = new AuthRepository();
    }

    public void signInWithGoogle(AuthCredential googleAuthCredential) {
        authenticatedUserLiveData = authRepository.firebaseSignInWithGoogle(googleAuthCredential);
    }

    public void createUser(UserFirebase authenticatedUser) {
        createdUserLiveData = authRepository.createUserInFirestoreIfNotExists(authenticatedUser);
    }

//    public MutableLiveData<String> liveDataText = new MutableLiveData<String>();
//
//    public void getUser() {
//        liveDataText.setValue(authRepository.getUser());
//    }
}

File: app/src/main/java/com/nakano/stampcardmvvm/viewModel/AuthViewModel.kt
package com.nakano.stampcardmvvm.viewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.google.firebase.auth.AuthCredential
import com.nakano.stampcardmvvm.model.repository.AuthRepository
import kotlinx.coroutines.launch

class AuthViewModel(
    var repository: AuthRepository
) : ViewModel() {
    private val _isSuccess = MutableLiveData<Boolean>()
    val isSuccess: LiveData<Boolean> = _isSuccess
    private val _email = MutableLiveData<String>()
    var email: LiveData<String> = _email

    fun signInWithGoogle(googleAuthCredential: AuthCredential?) {
        viewModelScope.launch {
            val result = repository.signInWithGoogle(googleAuthCredential)
            _isSuccess.postValue(result.value)
        }
    }

    fun signInWithTwitter() {
        viewModelScope.launch {
            val result = repository.signInWithTwitter()
            _isSuccess.postValue(result.value)
        }
    }

    fun signInWithEmailAndPassword(email: String, password: String) {
        viewModelScope.launch {
            val result = repository.signInWithEmailAndPassword(email, password)
            _isSuccess.postValue(result.value)
        }
    }

    fun createInWithEmailAndPassword(email: String, password: String) {
        viewModelScope.launch {
            val result = repository.createUserWithEmailAndPassword(email, password)
            _isSuccess.postValue(result.value)
        }
    }

    fun signInAnonymous() {
        viewModelScope.launch {
            val result = repository.signInAnonymous()
            _isSuccess.postValue(result.value)
        }
    }

    fun updateEmail(currentUserEmail: String, afterChangeEmail: String, password: String) {
        viewModelScope.launch {
            val result = repository.updateEmail(currentUserEmail, afterChangeEmail, password)
            _isSuccess.postValue(result.value)
        }
    }

    fun sendPasswordResetEmail(email: String) {
        viewModelScope.launch {
            val result = repository.sendPasswordResetEmail(email)
            _isSuccess.postValue(result.value)
        }
    }

    fun saveTmpEmail(email: String) {
        val result = repository.saveTmpEmail(email)
        _email.postValue(result.value)
    }

    fun getTmpEmail() {
        val result = repository.getTmpEmail()
        _email.postValue(result.value)
    }
}